EnrichmentFormSample/EnrichmentFormSample/Scripts/blades/edit-address.js

```diff
@@ -0,0 +1,245 @@
+                            break;
+                        case 'City':
+                            value = data.city;
+                            break;
+                        case 'Position':
+                            value = data.position ? data.position.lat.toString() + ',' + data.position.lng.toString() : data.position;
+                            break;
+                        case 'Zip':
+                            value = data.zip;
+                            break;
+                        default:
+                            continue;
+                    }
+                    if (property.values.length) {
+                        property.values[0].value = value;
+                    } else {
+                        property.values.push({ value: value, isInherited: false });
+                    }
+                }
+                blade.currentEntity.properties = properties;
+                $scope.bladeClose();
+            };
+
+            $scope.isValid = false;
+
+            function initAutocomplete() {
+                // Create the autocomplete object, restricting the search predictions to
+                // geographical location types.
+                // 
+                autocomplete = new google.maps.places.Autocomplete(
+                    document.getElementById('address_autocomplete'));
+
+                // Avoid paying for data that you don't need by restricting the set of
+                // place fields that are returned to just the address components.
+                autocomplete.setFields(['address_component', 'geometry']);
+
+                // When the user selects an address from the drop-down, populate the
+                // address fields in the form.
+                autocomplete.addListener('place_changed', function () {
+                    fillInAddress(autocomplete.getPlace());
+                });
+            }
+
+            function onMarkerDropped(event) {
+                geocoder.geocode({ 'latLng': marker.getPosition() }, function (results, status) {
+                    if (status === google.maps.GeocoderStatus.OK) {
+                        if (results[0]) {
+                            fillInAddress(results[0]);
+                            document.getElementById('address_autocomplete').value = results[0].formatted_address;
+                        }
+                    }
+                });
+            }
+
+            function setMarker(position) {
+                if (!marker) {
+                    marker = new google.maps.Marker({
+                        position: position,
+                        map: map,
+                        draggable: true
+                    });
+                    marker.addListener('dragend', onMarkerDropped);
+                } else {
+                    marker.setPosition(position);
+                }
+            }
+
+            function fillInAddress(place) {
+                // clear data
+                data.city = '';

```

Comment:
could it be data = {}; ?

Reply:
$scope.addressFields = addressFields = {}